File: backend/utils/analytics_events.py
"""Compatibility labels for analytics call sites that are no-ops in this edition.

Keeping labels mechanical avoids publishing the hosted product-event catalogue.
"""


class _EventLabels(type):
    def __getattr__(cls, name: str) -> str:
        return name.lower()


class Events(metaclass=_EventLabels):
    pass

File: frontend/app/lib/analytics-events.ts
/**
 * Analytics calls are compatibility no-ops in the community build. Resolve a
 * stable label mechanically so the hosted product-event catalogue is not part
 * of this source tree and shared call sites do not need edition branches.
 */
export const EVENTS: Record<string, string> = new Proxy({}, {
    get: (_target, property) => String(property).toLowerCase(),
});

export type AnalyticsEvent = string;
